### ISSUE TITLE
"Unwrap `Result` return type" shouldn't trigger in trait impls

### ISSUE THREAD

**Issue by jonas-schievink** [rust-lang/rust-analyzer#13560]
```rust
trait Tr {
    fn f() -> Result<(), ()>;
}

impl Tr for () {
    fn f() -> $0Result<(), ()> {
        todo!()
    }
}
```

This offers the "Unwrap `Result` return type", which removes the return type here, but this is wrong in a trait impl if the trait isn't also modified (which requires fixing up every impl of it).

**Comment by jonas-schievink** [comment]
Hmm, the same goes for "Add this function's return type". Now I'm not sure what the best thing to do here is, maybe we just want to trigger all signature-changing assists in trait impls and have the user do fixups?
